templates/master/00-master/_base/files/usr-local-bin-openshift-recovery-tools-sh.yaml

```diff
@@ -0,0 +1,313 @@
+filesystem: "root"
+mode: 644
+path: "/usr/local/bin/openshift-recovery-tools"
+contents:
+  inline: |
+    #!/usr/bin/env bash
+    
+    init() {
+      ASSET_BIN=${ASSET_DIR}/bin
+      if [ ! -d "$ASSET_BIN" ]; then
+        echo "Creating asset directory ${ASSET_DIR}"
+        for dir in {bin,tmp,shared,backup,templates,restore,manifests}; do
+          /usr/bin/mkdir -p ${ASSET_DIR}/${dir}
+        done
+      fi
+    }
+    
+    # download and test etcdctl from upstream release assets
+    dl_etcdctl() {
+      GOOGLE_URL=https://storage.googleapis.com/etcd
+      DOWNLOAD_URL=${GOOGLE_URL}
+    
+      echo "Downloading etcdctl binary.."
+      curl -s -L ${DOWNLOAD_URL}/${ETCD_VERSION}/etcd-${ETCD_VERSION}-linux-amd64.tar.gz -o $ASSET_DIR/tmp/etcd-${ETCD_VERSION}-linux-amd64.tar.gz \
```

Comment:
quick question: is my understanding correct that this tar file together with the manifest files which are pulled from other github repos will be baked inside the MCO image when a new release is cut and as such will all (including other operators) be wrapped in the CVO ?

Reply:
Not sure I am following completely. When this function is called [1] it will download a local copy of etcdctl into the callers local ~./assets/bin this binary will be used by various other functions for disaster recovery. So the tar itself is not baked in only the function to download it. The tar and etcdctl binary will not live on the cluster unless`dl_etcdctl` function is called. IN an upgrade $ETCD_VERSION may integrate. We will document that the asset dir should be archived after the restore is complete. 

[1] https://github.com/openshift/machine-config-operator/blob/3b266bed1fbb31f5f608c33ad87c8b7d3ccf1472/templates/master/00-master/_base/files/usr-local-bin-etcd-member-recover-sh.yaml#L52

Reply:
@hexfusion thanks for taking the time to respond 👍 

I'm still new in trying to get my head around how the whole solution is built and package.

The full context for my ask is around: how will this work in a air-gap/ disconnected environment w/o internet access ? 

if is not in scope then please ignore me

Reply:
>The full context for my ask is around: how will this work in a air-gap/ disconnected environment w/o internet access ?

In general, this is the first iteration of disaster recovery and we will improve on this concept until we get to a point where an operator handles these tasks autonomously. So yes I expect there to be areas for improvement and corner cases to solve. But we wanted to provide some tooling to help in the restoration process. Airgap DR would be a good use case to solve for in 4.2.

Reply:
fair point, thanks for feedback @hexfusion (i don't have full picture and so excuse my silly questions)